pkg/gofr/http/router.go

```diff
@@ -36,6 +47,95 @@ func (rou *Router) Add(method, pattern string, handler http.Handler) {
 	rou.Router.NewRoute().Methods(method).Path(pattern).Handler(h)
 }
 
+func (rou *Router) GetDefaultStaticFilesConfig() StaticFileConfig {
+	staticConfig := StaticFileConfig{
+		DirectoryListing: true,
+		HideDotFiles:     true,
+	}
+
+	return staticConfig
+}
+
+// Static File Handling.
+func (rou *Router) AddStaticFiles(endpoint string, staticConfig StaticFileConfig) {
+	fileServer := http.FileServer(http.Dir(staticConfig.FileDirectory))
+	rou.Router.NewRoute().PathPrefix(endpoint + "/").Handler(http.StripPrefix(endpoint, staticConfig.staticHandler(fileServer)))
+}
+
+// Check all the static handling configs.
+const forbiddenBody string = "403 forbidden"
+
+func (staticConfig StaticFileConfig) staticHandler(fileServer http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		url := r.URL.Path
+
+		if staticConfig.DirectoryListing {
+			staticConfig.checkDirectoryListing(w, r, url)
+		}
+
+		filePath := strings.Split(url, "/")
+
+		fileName := filePath[len(filePath)-1]
+
+		staticConfig.checkDotFiles(w, fileName, url)
+
+		if len(staticConfig.ExcludeExtensions) > 0 {
+			staticConfig.checkExcludedExtensions(w, fileName, url)
+		}
+
+		if len(staticConfig.ExcludeFiles) > 0 {
+			staticConfig.checkExcludedFiles(w, fileName, url)
+		}
+
+		fileServer.ServeHTTP(w, r)
+	})
+}
+
+func (staticConfig StaticFileConfig) checkDirectoryListing(w http.ResponseWriter, r *http.Request, url string) {
+	if _, err := os.Stat(filepath.Join(staticConfig.FileDirectory, "index.html")); err != nil && strings.HasSuffix(url, "/") {
+		http.NotFound(w, r)
+		return
+	}
+}
+
+func (staticConfig StaticFileConfig) checkDotFiles(w http.ResponseWriter, fileName, url string) {
+	if _, err := os.Stat(filepath.Join(staticConfig.FileDirectory, url)); err == nil && strings.HasPrefix(fileName, ".") {
```

Comment:
I thought about it while writing #156

I don't think checkDotFiles check for the name of the directory to check if the file is not in a hidden folder

Think about `/foo/.bar/image.jpg`

Reply:
Hmm, something to work upon then. I will split the request and check if any of them are a hidden folder or hidden file then.